### PR TITLE
[4/x][Chonker] Create prizes teams components

### PR DESCRIPTION
**Pull request by jzeiders**
I'm sorry, full stack for Project Teams.
**Enables:**
- Joining Teams
- Creating Teams
- Updating Metadata on Teams
- Removing Team Members
- Adding / Removing Prizes

**Also Includes:**
- Generic Component for multi-text forms

The code kinda whack will refactor at another point. 

[Demo](https://drive.google.com/open?id=1mSMrLvrS56t2MuMMlxspDl305OSQi-HO)

### CODE REVIEW

File: api/projectTeam.js [hunk 1]
```diff
@@ -6,15 +6,19 @@ const router = express.Router();
 router.use(utils.authMiddleware);
 router.use(utils.preprocessRequest);
 
+const getProjectTeamForSelf = req => {
+  const { id } = req.user;
+
+  return models.Person.findByPk(id, {
+    include: [{ model: models.ProjectTeam, required: true }]
+  }).then(p => p.ProjectTeam);
+};
```
Comment (saresend) [inline api/projectTeam.js]: can we use `await/async`, just to be consistent here?

File: api/projectTeam.js [hunk 2]
```diff
@@ -23,13 +27,13 @@ router.get("/self", async (req, res) => {
 router.put("/join/:name", async (req, res) => {
   try {
     const projectTeam = await models.ProjectTeam.findOne({
-      where: { name: req.params.name },
-      include: [models.Person]
+      where: { name: req.params.name }
     });
-    if (projectTeam.People.length >= 4) {
+    if (projectTeam.Members.length >= 4) {
       throw new Error("Team is full");
```
Comment (saresend) [inline api/projectTeam.js]: Wait, if this is a route, can we just `return res.status(400).json({ /* error msg */})`

File: api/projectTeam.js [hunk 3]
```diff
@@ -23,13 +27,13 @@ router.get("/self", async (req, res) => {
...
     }
-    projectTeam.addPerson(req.user.id);
+    await projectTeam.addMember(req.user.id);
+    await projectTeam.reload();
```
Comment (saresend) [inline api/projectTeam.js]: does `addMember` not already in-place update the projectTeam?

File: api/projectTeam.js [hunk 4]
```diff
@@ -66,15 +70,71 @@ router.post("/self", async (req, res) => {
...
+  } catch (e) {
+    return res.status(400).json({ err: e.message });
+  }
+});
+
+router.put("/self", async (req, res) => {
+  try {
+    const { body } = req;
+    const projectTeam = await getProjectTeamForSelf(req);
+    await projectTeam.update(body);
```
Comment (saresend) [inline api/projectTeam.js]: If this is a mutable route hackers can access, we need input validation.

File: api/projectTeam.js (hunk 3 above)
Comment (jzeiders) [reply api/projectTeam.js]: It wasn't not sure why

**Pull request merged**